lib/src/arithmetic/floating_point/floating_point_sqrt.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Indentifier: BSD-3-Clause
+//
+// floating_point_sqrt.dart
+// An abstract base class defining the API for floating-point square root.
+//
+// 2025 March 3
+// Authors: James Farwell <james.c.farwell@intel.com>,
+//Stephen Weeks <stephen.weeks@intel.com>,
+//Curtis Anderson <curtis.anders@intel.com>
+
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An abstract API for floating point square root.
+abstract class FloatingPointSqrt<FpType extends FloatingPoint> extends Module {
+  /// Width of the output exponent field.
+  final int exponentWidth;
+
+  /// Width of the output mantissa field.
+  final int mantissaWidth;
+
+  /// The [clk] : if a non-null clock signal is passed in, a pipestage is added
+  /// to the square root to help optimize frequency.
```

Comment:
I don't see the `clk` and `reset` used anywhere?  No sequentials  in here (yet?)? Did you mean to use it or is this plumbed for future upgrades?

Reply:
This is to make future work easier. In most cases, like all ALUs, you get your best bang for buck when it is pipelined. I wasn't sure how to do that yet, but I wanted to have the ground work. It also keeps it consistent with the other FloatingPoint modules that are also capable of pipeline features. So setup for future work to make it a bit easier (and once I figure out how to do it in ROHD).

Reply:
Ok, if we get this PR to a mergeable state except for pipelining, we should at least clearly document that the signals don't do anything yet and are plumbed for future pipelining, or pull the delta into some new branch for future work.  Ideally the users can read the doc comments to understand how the component works without looking inside to see what happens :)

Reply:
Added docs in the simple class clarifying this isn't implemented and meant for future work